src/builder.rs

```diff
@@ -737,15 +887,83 @@ fn pad_spend(
 ///
 /// The returned bundle will have no proof or signatures; these can be applied with
 /// [`Bundle::create_proof`] and [`Bundle::apply_signatures`] respectively.
+#[cfg(feature = "circuit")]
 #[allow(clippy::type_complexity)]
 pub fn bundle<V: TryFrom<i64>, FL: OrchardFlavor>(
-    mut rng: impl RngCore,
+    rng: impl RngCore,
     anchor: Anchor,
     bundle_type: BundleType,
     spends: Vec<SpendInfo>,
     outputs: Vec<OutputInfo>,
     burn: HashMap<AssetBase, NoteValue>,
 ) -> Result<UnauthorizedBundleWithMetadata<V, FL>, BuildError> {
+    build_bundle(
+        rng,
+        anchor,
+        bundle_type,
+        spends,
+        outputs,
+        burn,
+        |pre_actions, flags, value_balance, burn_vec, bundle_meta, mut rng| {
```

Comment:
`burn_vec` can be a slice here `&[(_,_)]`, for simplicity. (optional)

Reply:
I prefer to keep the vec. Otherwise, we would first convert the vec into a slice &[] (I think that has almost no cost). Then, we would convert the slice back into a vec to add it to the Bundle struct which would result in copying the data.

Reply:
I see, let's keep the `vec<>`.